<?xml version="1.0" encoding="UTF-8" standalone="no" ?>
<launch>

	<arg name="name" 		default="openGo" />
	<arg name="insole_freq" default="100" />
	<arg name="filename" 		default="/srv/host_data/ruoli/ViconData/Ruoli/Moticon_insole/RealTimekIDS2/walking01_header_corrected.txt"/>
	<arg name="estimated_delay"	default="0"/>
	<arg name="foot_length" 	default="0.2742"/>
	<arg name="foot_width" 		default="0.0975"/>
	<arg name="grf_origin_z_offset" default="0.0"/>
	<arg name="start_time"		default="0.0"/>

	<arg name="use_synchronization_event"				default="false"/>
	<arg name="rsecs"				default=""/>
	<arg name="rnsecs"				default=""/>
	<arg name="rto"				default=""/>
	<arg name="lsecs"				default=""/>
	<arg name="lnsecs"				default=""/>
	<arg name="lto"				default=""/>

	<arg name="tf_prefix" 				default="ik/"/>
	<node pkg="moticon_insoles" type="from_file.py" 	name="moticon_insoles" output="screen">
		<param name="name" 		value="$(arg name)" />
		<param name="insole_rate" 		value="$(arg insole_freq)" />
		<param name="left_cop_reference_frame" 			value="$(arg tf_prefix)left_insole" />
		<param name="right_cop_reference_frame" 		value="$(arg tf_prefix)right_insole" />
		<param name="foot_length" 				value="$(arg foot_length)"/>
		<param name="foot_width" 				value="$(arg foot_width)"/>
		<param name="wait_for_trigger" 				value="true"/>
		<param name="estimated_delay"				value="$(arg estimated_delay)"/>
		<param name="filename" 					value="$(arg filename)"/>
		<param name="grf_origin_z_offset" 			value="$(arg grf_origin_z_offset)"/>
		<param name="start_time"				value="$(arg start_time)"/>
		<param name="use_synchronization_event"			value="$(arg use_synchronization_event)"/>

		<param name="rsecs"	 value="$(arg rsecs)"	/>
		<param name="rnsecs"	 value="$(arg rnsecs)"	/>
		<param name="rto"	 value="$(arg rto)"		/>
		<param name="lsecs"	 value="$(arg lsecs)"	/>
		<param name="lnsecs"	 value="$(arg lnsecs)"	/>
		<param name="lto"	 value="$(arg lto)"		/>	
	</node>
	<include file="$(find moticon_insoles)/launch/imus_from_insole.launch"/>
	
</launch>


